fontir/src/variations.rs

```diff
@@ -0,0 +1,632 @@
+                axis_name
+            })
+            .cloned()
+            .collect();
+        let mut unordered_axes: Vec<_> = location
+            .axis_names()
+            .filter(|axis_name| !ordered_axes.contains(axis_name))
+            .map(|axis_name| {
+                // assign 0x10000 to any unknown axis
+                known_axes.push(0x10000);
+                axis_name
+            })
+            .cloned()
+            .collect();
+        unordered_axes.sort();
+        ordered_axes.extend(unordered_axes);
+
+        let axis_value_signs = ordered_axes
+            .iter()
+            .map(|axis_name| {
+                location
+                    .get(axis_name)
+                    .map(|coord| {
+                        let value = coord.into_inner();
+                        if value > ZERO {
+                            1i8
+                        } else if value < ZERO {
+                            -1i8
+                        } else {
+                            0i8
+                        }
+                    })
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        let axis_value_abs = ordered_axes
+            .iter()
+            .map(|axis_name| {
+                location
+                    .get(axis_name)
+                    .map(|coord| OrderedFloat(coord.into_inner().abs()))
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        return LocationSortKey {
+            rank,
+            on_axis_points,
+            known_axes,
+            ordered_axes,
+            axis_value_signs,
+            axis_value_abs,
+        };
+    }
+}
+
+/// Sort key for a location.
+///
+/// <https://github.com/fonttools/fonttools/blob/2f1f5e5e7be331d960a0e30d537c2b4c70d89285/Lib/fontTools/varLib/models.py#L326>
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct LocationSortKey {
+    // 1 for every non-zero entry in the map
+    rank: usize,
+    // -1 for every axis:value entry in location where a location exists whose only non-zero
+    // position is on axis at value. That is, a location exists that is directly on the axis there.
+    on_axis_points: i16,
+    // index in user-specified axis order, otherwise score 0x10000 (65536)
+    known_axes: Vec<usize>,
+    // ordered list of axis in location
+    ordered_axes: Vec<String>,
+    // signs of positions on axes in ordered_axes order
+    axis_value_signs: Vec<i8>,
+    // abs values of positions on axes in ordered_axes order
+    axis_value_abs: Vec<OrderedFloat<f32>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+struct VariationRegion(HashMap<String, Tent>);
+
+impl VariationRegion {
+    fn new() -> Self {
+        VariationRegion(HashMap::new())
+    }
+}
+
+impl VariationRegion {
+    /// The scalar multiplier for the provided location for this region
+    ///
+    /// Returns None for no influence, Some(non-zero value) for influence.
```

Comment:
why return an Option? I think supportScalar always returns a float between 0.0 (when location falls outside the tent) and +1.0 (at least with extrapolate=False); plus assumes per-axis scalar of 1.0 when peak is at 0.0 (ot=True), as in "axis does not participate".
I don't see what the None buys us

Reply:
ok, I see, you use None to mean scalar==0.0 so you can use Option methods like map() and similar instead of explicitly testing if scalar == 0.0

Reply:
on the other end, when you are actually _scaling_ the deltas, if you have something that's unconditionally a float then you can simply multiply that, and if scalar is 0, the delta is zeroed accordingly.. whereas with this Option you'd have to branch.

Reply:
I thought it would make the code simpler but I don't feel tremendously strongly about it, we could just return 0.

Reply:
It now returns 0